geps/gep-1494/index.md

```diff
@@ -0,0 +1,59 @@
+# GEP-1494: Auth in Gateway API
+
+* Issue: [#1494](https://github.com/kubernetes-sigs/gateway-api/issues/1494)
+* Status: Provisional
+
+(See status definitions [here](/geps/overview/#gep-states).)
+
+## TLDR
+
+Allow authentication works as a policy that can  be attached to both the Gateway and HTTPRoute.
+
+## Goals
+
+1). In the default policy, it can add an ExtensionFilter that specifies the auth so that Gateway can be attached to.
+2). Individual HTTPRoutes can choose to opt out by specifying an empty extension, or a null extension. 
+3). It can also have disabled authentication.
```

Comment:
Again, I think it's important to focus on the user story rather than details.

The thing that users (well, HTTPRoute owners) want to be able to do is to ask their Gateway API implementation to ensure that requests matching some pattern (such as a path like `/admin`) all have auth added if not present before requests are allowed to be forwarded to the backend.

So I think that we need a very high-level definition of that goal in here as the first item.

I think the second point is a really good one - that HTTPRoute owners must also be able to require that certain route rules do _not_ require authentication.

Reply:
I think opting out (at the level of HTTPRoutes) on a config otherwise enforced as a default (set at the level of the Gateway) is a need that extrapolates the auth domain, but rather a pattern that we may see at any kind of hierarchical config.

I would dedicate some careful thought on this, possibly out of the scope of this GEP.

Prior art: Kuadrant specifies the concept of [_Unsetting_](https://docs.kuadrant.io/0.8.0/architecture/rfcs/0009-defaults-and-overrides/#unsetting-inherited-defaults) for its AuthPolicy and RateLimitPolicy kinds.

Reply:
> I think opting out (at the level of HTTPRoutes) on a config otherwise enforced as a default (set at the level of the Gateway) is a need that extrapolates the auth domain, but rather a pattern that we may see at any kind of hierarchical config.
> 
> I would dedicate some careful thought on this, possibly out of the scope of this GEP.
> 
> Prior art: Kuadrant specifies the concept of [_Unsetting_](https://docs.kuadrant.io/0.8.0/architecture/rfcs/0009-defaults-and-overrides/#unsetting-inherited-defaults) for its AuthPolicy and RateLimitPolicy kinds.

My proposal is to implement one of the following options for authentication (AuthN): a default policy, an override policy, or a disabled policy. I want to keep it simple, but I’m wondering if I should consider making it more complex...